sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -0,0 +1,372 @@
+"""Adapter for using huggingface transformers for forecasting."""
+
+import numpy as np
+import pandas as pd
+import torch
+from torch.utils.data import Dataset
+from transformers import AutoConfig, Trainer, TrainingArguments
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+__author__ = ["benheid"]
+
+
+class HFTransformersForecaster(BaseForecaster):
+    """
+    Forecaster that uses a huggingface model for forecasting.
+
+    This forecaster fetches the model from the huggingface model hub.
+    Note, this forecaster is in an experimental state. It is currently only
+    working for Informer, Autoformer, and TimeSeriesTransformer. 
+
+    Parameters
+    ----------
+    model_path : str
+        Path to the huggingface model to use for forecasting
+    fit_strategy : str, default="minimal"
+        Strategy to use for fitting the model. Can be "minimal" or "full"
+    validation_split : float, default=0.2
+        Fraction of the data to use for validation
+    config : dict, default={}
+        Configuration to use for the model.
```

Comment:
Where do we get the valid options? I don't think this is available for all models.

Reply:
This depends on the model. I added a statement that you should take a look at the model card.